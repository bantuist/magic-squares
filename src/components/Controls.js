import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  justify-content: center;
`;
const Button = styled.button`
  cursor: pointer;
  margin: 0 5px;
`;

const Controls = ({ onSolve, onClue, onReset }) => {
  return (
    <Container>
      <Button className="solve-button" onClick={onSolve}>Solve</Button>
      <Button className="clue-button" onClick={onClue}>Clue</Button>
      <Button className="reset-button" onClick={onReset}>Reset</Button>
    </Container>
  );
};

export default Controls;