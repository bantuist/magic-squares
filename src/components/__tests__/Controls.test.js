import React from 'react';
import { shallow } from 'enzyme';
import Controls from '../Controls';

const onSolve = jest.fn();
const onClue = jest.fn();
const onReset = jest.fn();
const props = { onSolve, onClue, onReset };

describe('Controls', () => {
  const controls = shallow(<Controls { ...props }/>);

  it('renders without crashing', () => {
    expect(controls).toMatchSnapshot();
  });

  it('calls `onSolve()`', () => {
    controls.find('.solve-button').simulate('click');
    expect(props.onSolve).toHaveBeenCalled();
  });

  it('calls `onClue()`', () => {
    controls.find('.clue-button').simulate('click');
    expect(props.onSolve).toHaveBeenCalled();
  });

  it('calls `onReset()`', () => {
    controls.find('.reset-button').simulate('click');
    expect(props.onSolve).toHaveBeenCalled();
  });
});
