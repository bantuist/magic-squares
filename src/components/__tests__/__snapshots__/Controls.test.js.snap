// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls
    onClue={[Function]}
    onReset={[Function]}
    onSolve={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.button
          className="solve-button"
          onClick={[Function]}
>
          Solve
</styled.button>,
        <styled.button
          className="clue-button"
          onClick={[Function]}
>
          Clue
</styled.button>,
        <styled.button
          className="reset-button"
          onClick={[Function]}
>
          Reset
</styled.button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Solve",
          "className": "solve-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Solve",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Clue",
          "className": "clue-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Clue",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Reset",
          "className": "reset-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Reset",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.button
            className="solve-button"
            onClick={[Function]}
>
            Solve
</styled.button>,
          <styled.button
            className="clue-button"
            onClick={[Function]}
>
            Clue
</styled.button>,
          <styled.button
            className="reset-button"
            onClick={[Function]}
>
            Reset
</styled.button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Solve",
            "className": "solve-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Solve",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Clue",
            "className": "clue-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Clue",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Reset",
            "className": "reset-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reset",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
